brainsets_pipelines/kemp_sleep_edf_2013/pipeline.py

```diff
@@ -0,0 +1,448 @@
+
+        self.update_status("PROCESSING")
+
+        base_name = psg_path.stem
+
+        output_path = self.processed_dir / f"{base_name}.h5"
+        if output_path.exists() and not self.args.reprocess:
+            self.update_status("Skipped Processing")
+            logging.info(f"Skipping processing, file exists: {output_path}")
+            return
+
+        brainset_description = BrainsetDescription(
+            id="kemp_sleep_edf_2013",
+            origin_version="1.0.0",
+            derived_version="1.0.0",
+            source="https://www.physionet.org/content/sleep-edfx/1.0.0/",
+            description="Sleep-EDF Database Expanded containing 197 whole-night "
+            "polysomnographic sleep recordings with EEG, EOG, EMG, and sleep stage annotations.",
+        )
+
+        logging.info(f"Processing file: {psg_path}")
+
+        self.update_status("Loading EDF")
+        raw_psg = mne.io.read_raw_edf(str(psg_path), preload=True, verbose=False)
+
+        self.update_status("Extracting Metadata")
+        age, sex = parse_subject_metadata(raw_psg)
+
+        if "SC4" in base_name:
+            subject_id = base_name[3:5]
+            study_type = "sleep_cassette"
+        elif "ST7" in base_name:
+            subject_id = base_name[3:5]
+            study_type = "sleep_telemetry"
+        else:
+            subject_id = base_name
+            study_type = "unknown"
+
+        subject = SubjectDescription(
+            id=f"{study_type}_{subject_id}",
+            species=Species.HOMO_SAPIENS,
+            age=age,
+            sex=sex,
+        )
+
+        recording_date = raw_psg.info.get("meas_date")
+        if recording_date is not None:
+            recording_date = recording_date.strftime("%Y-%m-%d")
+
+        session_description = SessionDescription(
+            id=base_name,
+            recording_date=recording_date,
+        )
+
+        device_description = DeviceDescription(
+            id=base_name,
+            recording_tech=RecordingTech.POLYSOMNOGRAPHY,
+        )
+
+        self.update_status("Extracting Signals")
+        signals, units = extract_signals(raw_psg)
+
+        if signals is None:
+            logging.error("No signals extracted from PSG file")
+            self.update_status("FAILED")
+            return
+
+        stages = None
+        if hypnogram_path and hypnogram_path.exists():
+            self.update_status("Extracting Sleep Stages")
+            stages = extract_sleep_stages(str(hypnogram_path), raw_psg.times)
+
+        splits = None
+        if stages is not None:
+            self.update_status("Creating Splits")
+            splits = create_splits(stages, n_folds=3, seed=42)
+
+        data_dict = {
+            "brainset": brainset_description,
+            "subject": subject,
+            "session": session_description,
+            "device": device_description,
+            "eeg": signals,
+            "units": units,
+            "domain": signals.domain,
+        }
```

Comment:
Very nitpicky, but why not follow the convention of adding brainset, subject, session, device, eeg and units directly into the Data object (instead of everything into a Dict())? Only enclose behavioural variables into a Dict() and then unpack it